src/services/pallets/PalletsErrorsService.ts

```diff
@@ -0,0 +1,298 @@
+		} else if (onlyIds) {
+			items = Object.entries(errors).map(
+				(errorItem) => errorItem[0] as unknown as Text
+			);
+		} else {
+			items = [] as ErrorMetadataLatest[];
+			for (const [, value] of Object.entries(errors)) {
+				const item = value.meta;
+				items.push(item);
+			}
+		}
+
+		return {
+			at: {
+				hash: hash,
+				height: number.unwrap().toString(10),
+			},
+			pallet: stringCamelCase(palletMeta.name),
+			palletIndex: palletMetaIdx,
+			items,
+		};
+	}
+
+	/**
+	 * Normalize error item metadata by running it through `sanitizeNumbers` and
+	 * converting the docs section from an array of strings to a single string
+	 * joined with new line characters.
+	 *
+	 * @param errorItemMeta polkadot-js ErrorEntryMetadataV12
+	 */
+	private normalizeErrorItemMeta(
+		errorItemMeta: ErrorMetadataLatest
+	): ISanitizedErrorItemMetadata {
+		const normalizedErrorItemMeta = sanitizeNumbers(
+			errorItemMeta
+		) as unknown as ISanitizedErrorItemMetadata;
+
+		normalizedErrorItemMeta.docs = this.sanitizeDocs(errorItemMeta.docs);
+
+		return normalizedErrorItemMeta;
+	}
+
+	/**
+	 * Find the error item's metadata within the pallets's metadata.
+	 *
+	 * @param palletMeta the metadata of the pallet that contains the error item
+	 * @param errorId name of the error item in camel or pascal case
+	 */
+	private findErrorItemMeta(
+		palletMeta: PalletMetadataV14,
+		errorItemId: string,
+		errors: ModuleErrors<'promise'>
+	): PalletErrorMetadataV14 {
+		if (palletMeta.errors.isNone) {
+			throw new InternalServerError(
+				`No error items found in ${palletMeta.name.toString()}'s metadata`
+			);
+		}
+
+		let errorItemMetaIdx = -1;
+		for (const [, value] of Object.entries(errors)) {
+			if (value.meta.name.toLowerCase() === errorItemId.toLowerCase()) {
+				errorItemMetaIdx = value.meta.index.toNumber();
+			}
+		}
+
+		if (errorItemMetaIdx === -1) {
+			throw new InternalServerError(
+				`Could not find error item ("${errorItemId}") in metadata. Error item names are expected to be in camel case, e.g. 'errorItemId'`
+			);
+		}
+
+		return Object.entries(errors)[
+			errorItemMetaIdx
+		] as unknown as PalletErrorMetadataV14;
+	}
+
+	/**
+	 * Find a pallet's metadata info.
+	 *
+	 * @param palletId identifier for a FRAME pallet as a pallet name or index.
+	 */
+	private findPalletMeta(
+		adjustedMetadata: MetadataV14,
+		historicApi: ApiDecoration<'promise'>,
+		palletId: string
+	): [PalletMetadataV14, number] {
+		const pallets: Vec<PalletMetadataV14> = adjustedMetadata['pallets'];
+		const filtered: PalletMetadataV14[] = pallets.filter(
+			(mod) => mod.errors.isSome
+		);
+
+		const { isValidPalletName, isValidPalletIndex, parsedPalletId } =
+			this.validPalletId(historicApi, pallets, palletId);
+
+		let palletMeta: PalletMetadataV14 | undefined;
+		let palletIdx: number | undefined;
+
+		if (isValidPalletIndex) {
+			palletIdx = parsedPalletId as number;
+			for (const [_sectionIdx, section] of filtered.entries()) {
```

Comment:
Whats the idea behind the `_` syntax here for just `sectionIdx`?

Reply:
I know we do the same thing in the `pallets storage` endpoint, but I think it is something that can be fixed and changed here.

Reply:
yea, mainly did this to be consistent with the `pallets storage` endpoint. Thinking we should just drop the `_`?

Reply:
Yea dropping the `_` will be idle. My thought being we will eventually fix the other one too. Once we get these pallet endpoints settled we'll go back to `pallets/storage` do some cleanup/rework